components/base/sensorcontrolled/movestraight.go

```diff
@@ -19,38 +19,45 @@ const (
 	headingGain           = 1.
 )
 
+// MoveStraight commands a base to move forward for the desired distanceMm at the given mmPerSec.
+// When controls are enabled, MoveStraight calculates the required velocity to reach mmPerSec
+// and the distanceMm goal. It then polls the provided velocity movement sensor and corrects any
+// error between this calculated velocity and the actual velocity using a PID control loop.
```

Comment:
@JohnN193 is this true? I see nothing about the velocity movement sensor but I assume it has to use that? might need to add a check for it

Reply:
Yeah, it uses the velocity sensor via the pid loop.

its also tracking position to stop moving when it reaches the goal, and attempts to keep the heading of the rover fixed if an appropriate sensor is provided

Reply:
The control loop isn’t configured unless velocity exists already

Reply:
right but we are checking in line 33 if position or control loop exists. what if you don't have a velocity movement sensor, so  no control loop, but you do have a position reporting sensor? we will run into issues where we try to create a control loop when we don't have a control loop config

Reply:
okay I understand now nvm

Reply:
I'll just add a comment